lift/Readme.md

```diff
@@ -6,271 +6,360 @@
 
-```ruby
-class ChangeUserName < ActiveRecord::Migration[5.0]
-  def change
-    reversible do |dir|
-      change_table :User do |t|
-        t.string :name, null: false
-      end
-    end
-  end
-end
-```
+**TODO**
 
-With imperative migrations, the user needs to both calculate the "diff" in his head and write the needed actions down, which will result in the new schema.
+</details>
 
-While in Prismas model the declarative datamodel is the source of truth for migrations, Active Record takes the imperative migrations as the source of truth for
-migrations. Therefore one can call the imperative migrations also **manual migrations**.
+### Migration
 
-### Datamodel Snapshots
+A Migration is a grouping of one or more Steps. Migrations run in a transaction if the datasource allows it.
 
-As mentioned in [Declarative migration](#how-does-naming-of-migrations-work), the user adjusts the `datamodel.prisma` file, which results in a new migration
-folder when running `prisma migrate`. In order for the migration engine to know which changes need to be applied, it needs a "copy" of the datamodel. As this
-"copy" describes the datamodel in a specific point in time, we also refer to it as a "snapshot".
+### Hook
 
-### Migration Engine
+A hook is a custom shell script that runs either before or after the migration. Hooks give you full control over your migrations. You can write hooks to:
 
-The migration engine is the heart of the whole migration system. It manages the connection to the database and performs the actual changes in the database
-schema. It is written in Rust and will be remotely controlled from the Prisma CLI written in JavaScript.
+- Ensure the data follows new constraints like `unique` or `non null`.
+- Add specific database primitives like functions in Postgres, which are not yet supported by Prisma.
+- Seed the database with data after a migration has been executed.
 
-### Migration History
+## Architecture
 
-When a migration is running or has been executed successfully, Prisma writes this information into a table or collection in the database. This table is also
-known as the "Migration History". It includes all successful and failed migrations of the specific Prisma Project. The benefits of storing this migration
-history are the following:
+Lift has 2 parts: a **Lift Client** and a **Lift Server**. By default, the Lift Server runs locally as a sidecar process. This makes it easy to get started. As
+your team's data requirements become more complex, you may prefer to handle your migrations on a remote machine where you have fine-grained access control over
+migrations.
 
-- Prisma can make sure, that the local migrations are in sync with the databases migrations
-- Prisma can ensure, that migrations don't get run twice
-- Prisma can give an overview in a GUI interface about the currently running migration and migrations that already have been running
-- Prisma can rollback migrations when there has been an error
+```
```

Comment:
Cosmetics: This diagram is a bit messed up on GH. Is this fixable?

Reply:
<img width="552" alt="Screen Shot 2019-09-27 at 11 06 38 AM" src="https://user-images.githubusercontent.com/170299/65757166-e50ee700-e116-11e9-89fb-e9a9e8dd7a84.png">

^ seems okay to me.